server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -449,13 +449,37 @@ public Settings getSettings() {
         return this.aliases;
     }
 
+    /**
+     * Return an object that maps each type to the associated mappings.
+     * The return value is never {@code null} but may be empty if the index
+     * has no mappings.
+     * @deprecated Use {@link #mapping()} instead now that indices have a single type
+     */
+    @Deprecated
     public ImmutableOpenMap<String, MappingMetaData> getMappings() {
```

Comment:
Note that we should remove this one too in the future but decided to keep it for a follow-up PR as this PR would otherwise be much larger.